Building hosted blazor wasm app in docker results in exception 'DefineStaticWebAssetEndpoints' with dotnet 9

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Updating a asp net hosted blazor wasm project to net 9 or creating a new from template(also net9) with the asp net building to docker. Attempting to build the host project with docker results in error DefineStaticWebAssetEndpoints task failing unexpectedly. It seems to be looking for missing .pdb file from the wasm project obj folder?

### Expected Behavior

The build completes successfully.

### Steps To Reproduce

Create a new wasm blazor asp net hosted app. 
Add docker support to the host project in visual studio
Build 'docker build -f .\BlazorApp2\Dockerfile .*

Below is the output from docker build. 

### Exceptions (if any)

[+] Building 15.3s (16/18)                                                                               docker:default
 => [internal] load build definition from Dockerfile                                                               0.0s
 => => transferring dockerfile: 1.35kB                                                                             0.0s
 => [internal] load metadata for mcr.microsoft.com/dotnet/sdk:9.0-preview                                          0.3s
 => [internal] load metadata for mcr.microsoft.com/dotnet/aspnet:9.0-preview                                       0.0s
 => [internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 464B                                                                                  0.0s
 => [build 1/8] FROM mcr.microsoft.com/dotnet/sdk:9.0-preview@sha256:3d8779247205d52c4f24e9e24a28451362244cde2831  0.0s
 => [base 1/2] FROM mcr.microsoft.com/dotnet/aspnet:9.0-preview                                                    0.0s
 => [internal] load build context                                                                                  0.1s
 => => transferring context: 12.98kB                                                                               0.1s
 => CACHED [base 2/2] WORKDIR /app                                                                                 0.0s
 => CACHED [build 2/8] WORKDIR /src                                                                                0.0s
 => [final 1/2] WORKDIR /app                                                                                       0.0s
 => [build 3/8] COPY [BlazorApp2/BlazorApp2.csproj, BlazorApp2/]                                                   0.0s
 => [build 4/8] COPY [BlazorApp2.Client/BlazorApp2.Client.csproj, BlazorApp2.Client/]                              0.0s
 => [build 5/8] RUN dotnet restore "./BlazorApp2/BlazorApp2.csproj"                                                8.3s
 => [build 6/8] COPY . .                                                                                           0.0s
 => [build 7/8] WORKDIR /src/BlazorApp2                                                                            0.0s
 => ERROR [build 8/8] RUN dotnet build "./BlazorApp2.csproj" -c Release -o /app/build                              6.5s
------
 > [build 8/8] RUN dotnet build "./BlazorApp2.csproj" -c Release -o /app/build:
1.075   Determining projects to restore...
1.478   All projects are up-to-date for restore.
1.532 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(314,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/src/BlazorApp2/BlazorApp2.csproj]
5.712   BlazorApp2.Client -> /app/build/BlazorApp2.Client.dll
5.714   BlazorApp2.Client (Blazor output) -> /app/build/wwwroot
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: The "DefineStaticWebAssetEndpoints" task failed unexpectedly. [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: System.IO.FileNotFoundException: Could not find file '/src/BlazorApp2/obj/Release/net9.0/BlazorApp2.Client.pdb'. [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: File name: '/src/BlazorApp2/obj/Release/net9.0/BlazorApp2.Client.pdb' [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at System.IO.FileInfo.get_Length() [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.GetFileLength(StaticWebAsset asset) [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.CreateEndpoint(StaticWebAsset asset, ContentTypeMapping[] contentTypeMappings) [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.Execute() [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute() [/src/BlazorApp2/BlazorApp2.csproj]
6.443 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/src/BlazorApp2/BlazorApp2.csproj]
6.450
6.451 Build FAILED.
6.451
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: The "DefineStaticWebAssetEndpoints" task failed unexpectedly. [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: System.IO.FileNotFoundException: Could not find file '/src/BlazorApp2/obj/Release/net9.0/BlazorApp2.Client.pdb'. [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018: File name: '/src/BlazorApp2/obj/Release/net9.0/BlazorApp2.Client.pdb' [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at System.IO.FileInfo.get_Length() [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.GetFileLength(StaticWebAsset asset) [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.CreateEndpoint(StaticWebAsset asset, ContentTypeMapping[] contentTypeMappings) [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.Execute() [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute() [/src/BlazorApp2/BlazorApp2.csproj]
6.451 /usr/share/dotnet/sdk/9.0.100-preview.5.24307.3/Sdks/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.References.targets(119,7): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/src/BlazorApp2/BlazorApp2.csproj]
6.451     0 Warning(s)
6.451     1 Error(s)
6.452
6.452 Time Elapsed 00:00:05.71
------
Dockerfile:19
--------------------
  17 |     COPY . .
  18 |     WORKDIR "/src/BlazorApp2"
  19 | >>> RUN dotnet build "./BlazorApp2.csproj" -c $BUILD_CONFIGURATION -o /app/build
  20 |
  21 |     # This stage is used to publish the service project to be copied to the final stage
--------------------
ERROR: failed to solve: process "/bin/sh -c dotnet build \"./BlazorApp2.csproj\" -c $BUILD_CONFIGURATION -o /app/build" did not complete successfully: exit code: 1

### .NET Version

net 9 preview 4/5 (didnt test before these)

### Anything else?

_No response_

--- a/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets
+++ b/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets
@@ -252,6 +252,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PropertyGroup>
       <_WasmBuildWebCilPath>$([MSBuild]::NormalizeDirectory($(IntermediateOutputPath), 'webcil'))</_WasmBuildWebCilPath>
       <_WasmBuildTmpWebCilPath>$([MSBuild]::NormalizeDirectory($(IntermediateOutputPath), 'tmp-webcil'))</_WasmBuildTmpWebCilPath>
+      <_WasmBuildOuputPath>$([MSBuild]::EnsureTrailingSlash('$(OutputPath)'))wwwroot\</_WasmBuildOuputPath>
     </PropertyGroup>
 
     <ConvertDllsToWebCil Candidates="@(_BuildAssetsCandidates)" IntermediateOutputPath="$(_WasmBuildTmpWebCilPath)" OutputPath="$(_WasmBuildWebCilPath)" IsEnabled="$(_WasmEnableWebcil)">
@@ -267,7 +268,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       AssetRole="Primary"
       CopyToOutputDirectory="PreserveNewest"
       CopyToPublishDirectory="Never"
-      ContentRoot="$(OutputPath)wwwroot"
+      ContentRoot="$(_WasmBuildOuputPath)"
       BasePath="$(StaticWebAssetBasePath)"
     >
       <Output TaskParameter="Assets" ItemName="WasmStaticWebAsset" />

--- a/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets
+++ b/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets
@@ -252,7 +252,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PropertyGroup>
       <_WasmBuildWebCilPath>$([MSBuild]::NormalizeDirectory($(IntermediateOutputPath), 'webcil'))</_WasmBuildWebCilPath>
       <_WasmBuildTmpWebCilPath>$([MSBuild]::NormalizeDirectory($(IntermediateOutputPath), 'tmp-webcil'))</_WasmBuildTmpWebCilPath>
-      <_WasmBuildOuputPath>$([MSBuild]::EnsureTrailingSlash('$(OutputPath)'))wwwroot\</_WasmBuildOuputPath>
+      <_WasmBuildOuputPath>$([MSBuild]::NormalizeDirectory('$(OutputPath)', 'wwwroot'))</_WasmBuildOuputPath>
     </PropertyGroup>
 
     <ConvertDllsToWebCil Candidates="@(_BuildAssetsCandidates)" IntermediateOutputPath="$(_WasmBuildTmpWebCilPath)" OutputPath="$(_WasmBuildWebCilPath)" IsEnabled="$(_WasmEnableWebcil)">

